pkg/cloud-controller-manager/loadBalancer.go

```diff
@@ -69,10 +71,10 @@ func (l *LoadBalancerManager) GetLoadBalancerName(ctx context.Context, clusterNa
 	suffix := fmt.Sprintf("%8x", digest)
 	name += suffix
 
-	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names
-	// The name contains no more than 253 characters
-	if len(name) > 253 {
-		name = name[:253]
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
```

Comment:
The name of a Service object must be a valid
[RFC 1035 label name](/docs/concepts/overview/working-with-objects/names#rfc-1035-label-names).